diff --git a/node_modules/expo-modules-core/android/src/main/class/expo/modules/kotlin/types/LazyKType.kt b/node_modules/expo-modules-core/android/src/main/class/expo/modules/kotlin/types/LazyKType.kt
new file mode 100644
index 0000000..d0168fe
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/class/expo/modules/kotlin/types/LazyKType.kt
@@ -0,0 +1,42 @@
+package expo.modules.kotlin.types
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeProjection
+
+class LazyKType(
+  override val classifier: KClass<*>,
+  override val isMarkedNullable: Boolean = false,
+  val kTypeProvider: () -> KType
+) : KType {
+  private var _kType: KType? = null
+  private val kType: KType
+    get() {
+      if (_kType == null) {
+        _kType = kTypeProvider()
+      }
+      return _kType!!
+    }
+
+  override val annotations: List<Annotation>
+    get() = kType.annotations
+  override val arguments: List<KTypeProjection>
+    get() = kType.arguments
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is LazyKType) return this.kType == other
+
+    return classifier == other.classifier && isMarkedNullable == other.isMarkedNullable
+  }
+
+  override fun hashCode(): Int {
+    var result = classifier.hashCode()
+    result = 31 * result + isMarkedNullable.hashCode()
+    return result
+  }
+
+  override fun toString(): String {
+    return kType.toString()
+  }
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/LazyKType.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/LazyKType.kt
new file mode 100644
index 0000000..d0168fe
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/LazyKType.kt
@@ -0,0 +1,42 @@
+package expo.modules.kotlin.types
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeProjection
+
+class LazyKType(
+  override val classifier: KClass<*>,
+  override val isMarkedNullable: Boolean = false,
+  val kTypeProvider: () -> KType
+) : KType {
+  private var _kType: KType? = null
+  private val kType: KType
+    get() {
+      if (_kType == null) {
+        _kType = kTypeProvider()
+      }
+      return _kType!!
+    }
+
+  override val annotations: List<Annotation>
+    get() = kType.annotations
+  override val arguments: List<KTypeProjection>
+    get() = kType.arguments
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is LazyKType) return this.kType == other
+
+    return classifier == other.classifier && isMarkedNullable == other.isMarkedNullable
+  }
+
+  override fun hashCode(): Int {
+    var result = classifier.hashCode()
+    result = 31 * result + isMarkedNullable.hashCode()
+    return result
+  }
+
+  override fun toString(): String {
+    return kType.toString()
+  }
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewDefinitionBuilder.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewDefinitionBuilder.kt
index c330026..1b10b25 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewDefinitionBuilder.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewDefinitionBuilder.kt
@@ -21,10 +21,12 @@ import expo.modules.kotlin.functions.createAsyncFunctionComponent
 import expo.modules.kotlin.modules.DefinitionMarker
 import expo.modules.kotlin.types.TypeConverterProvider
 import expo.modules.kotlin.types.descriptors.TypeDescriptor
+import expo.modules.kotlin.types.descriptors.toTypeDescriptor
 import expo.modules.kotlin.types.enforceType
 import expo.modules.kotlin.types.toAnyType
 import expo.modules.kotlin.types.toArgsArray
 import kotlin.reflect.KClass
+import kotlin.reflect.KType
 
 @DefinitionMarker
 class ViewDefinitionBuilder<T : View>(
@@ -32,6 +34,16 @@ class ViewDefinitionBuilder<T : View>(
   @PublishedApi internal val viewType: TypeDescriptor,
   @PublishedApi internal val converters: TypeConverterProvider? = null
 ) {
+  constructor(
+    viewClass: KClass<T>,
+    viewType: KType,
+    converters: TypeConverterProvider? = null
+  ) : this(
+    viewClass,
+    viewType.toTypeDescriptor(),
+    converters
+  )
+
   @PublishedApi
   internal var name: String = viewClass.simpleName ?: viewClass.java.simpleName
 
